contracts/Treasury.sol

```diff
@@ -174,21 +184,95 @@ contract OlympusTreasury is OlympusAccessControlled, ITreasury {
     }
 
     /**
-     * @notice mint new OHM using excess reserves
+     * @notice allow approved address to deposit an ERC721
+     * @param _token address
+     * @param _tokenId uint256
+     */
+    function depositERC721(address _token, uint256 _tokenId) external override {
+        IERC721(_token).safeTransferFrom(msg.sender, address(this), _tokenId);
+        emit DepositERC721(_token, _tokenId);
+    }
+
+    /**
+     * @notice allow approved address to withdraw ERC721
+     * @param _token address
+     * @param _tokenId uint256
+     */
+    function withdrawERC721(address _token, uint256 _tokenId) external override onlyGovernor {
+        IERC721 erc721 = IERC721(_token);
+        erc721.approve(msg.sender, _tokenId);
+        erc721.safeTransferFrom(address(this), msg.sender, _tokenId);
+
+        emit WithdrawERC721(_token, _tokenId);
+    }
+
+    /**
+     * @notice allocators can manage assets without being limited by excessReserves
+     * @notice always ensure the reserves are repalaced
+     * @param _token address
+     * @param _amount uint256
+     */
+    function allocatorManage(address _token, uint256 _amount) external override {
+        require(permissions[STATUS.ALLOCATOR][msg.sender], notApproved);
+
+        if (permissions[STATUS.RESERVETOKEN][_token] || permissions[STATUS.LIQUIDITYTOKEN][_token]) {
+            uint256 value = tokenValue(_token, _amount);
+            totalReserves = (value < totalReserves) ? totalReserves.sub(value) : 0;
+        }
+        IERC20(_token).safeTransfer(msg.sender, _amount);
+        emit AllocatorManaged(_token, _amount);
+    }
+
+    /**
+     * @notice Claim rewards from a Liquidity Staking vault on NFTX
+     * @param _liquidityStaking address
+     * @param _vaultId uint256
+     * @param _rewardToken address
+     */
+    function claimNFTXRewards(address _liquidityStaking, uint256 _vaultId, address _rewardToken) external override {
+        require(permissions[STATUS.ALLOCATOR][msg.sender], notApproved);
+        require(permissions[STATUS.RESERVETOKEN][_rewardToken], notAccepted);
+
+        // Get the reward token held in the treasury before our claim
+        uint256 previousBalance = IERC20(_rewardToken).balanceOf(address(this));
+
+        // Claim rewards from the NFTX vault
+        INFTXLPStaking(_liquidityStaking).claimRewards(_vaultId);
+
+        // Get our updated balance after claiming rewards
+        uint256 newBalance = IERC20(_rewardToken).balanceOf(address(this));
+
+        // If our balance has not changed, we don't need to process further
+        if (newBalance <= previousBalance) {
+            return;
+        }
+
+        uint256 balanceDifference = previousBalance - newBalance;
```

Comment:
This should be updated to `uint256 balanceDifference = newBalance.sub(previousBalance);`